client/components/jetpack/has-vaultpress-switch/index.tsx

```diff
@@ -9,46 +9,63 @@ import { useSelector } from 'react-redux';
  */
 import getSelectedSiteId from 'calypso/state/ui/selectors/get-selected-site-id';
 import getRewindState from 'calypso/state/selectors/get-rewind-state';
+import getSiteScanState from 'calypso/state/selectors/get-site-scan-state';
 import QueryRewindState from 'calypso/components/data/query-rewind-state';
+import QueryJetpackScan from 'calypso/components/data/query-jetpack-scan';
 import RenderSwitch from 'calypso/components/jetpack/render-switch';
 
-const rewindStateImpliesVaultPress = ( rewindState?: { state?: string; reason?: string } ) => {
-	if ( ! rewindState ) {
+const productStateImpliesVaultPress = ( productState?: { state?: string; reason?: string } ) => {
```

Comment:
I added queries for the Jetpack Scan product state to this component, to cover sites that may only have purchased Scan.